networkx/utils/backends.py

```diff
@@ -33,162 +33,254 @@
     H = nx_parallel.ParallelGraph(G)
     nx.betweenness_centrality(H, k=10)
 
-How it works: You might have seen the ``@nx._dispatchable`` decorator on
-many of the NetworkX functions in the codebase. It decorates the function
-with code that redirects execution to the function's backend implementation.
-The code also manages any ``backend_kwargs`` you provide to the backend
-version of the function. The code looks for the environment variable or
-a ``backend`` keyword argument and if found, converts the input NetworkX
-graph to the backend format before calling the backend's version of the
-function. If no environment variable or backend keyword are found, the
-dispatching code checks the input graph object for an attribute
-called ``__networkx_backend__`` which tells it which backend provides this
-graph type. That backend's version of the function is then called.
-The backend system relies on Python ``entry_point`` system to signal
-NetworkX that a backend is installed (even if not imported yet). Thus no
-code needs to be changed between running with NetworkX and running with
-a backend to NetworkX. The attribute ``__networkx_backend__`` holds a
-string with the name of the ``entry_point``. If none of these options
-are being used, the decorator code simply calls the NetworkX function
-on the NetworkX graph as usual.
+The first approach is useful when you don't want to change your nx code and just want
+to run your code on different backend(s). The second approach comes in handy when you
+need to pass additional backend-specific arguments, for example::
+
+    nx.betweenness_centrality(G, k=10, backend="parallel", get_chunks=get_chunks)
+
+Here, ``get_chunks`` is not a NetworkX argument, but a nx_parallel-specific argument.
+
+NetworkX also offers a very basic logging system that can help you verify if the
+backend that you specified is being implemented. This will most likely become more
+enhanced in the future. You can enable the networkx's backend logger like this::
+
+    import logging
+    logging.basicConfig(level=logging.DEBUG)
+    nx.betweenness_centrality(G, backend="parallel")
+    # Output
+    # DEBUG:networkx.utils.backends:using backend 'parallel' for call to `betweenness_centrality' with args: (Parallel<networkx.classes.graph.Graph object at 0x1008144d0>, None, True, None, False, <random.Random object at 0x135834c20>), kwargs: {}
+    # {1: 0.0, 2: 0.0, 3: 0.0, 4: 0.0, 5: 0.0, 6: 0.0}
+
+And you can disable all the loggers by running this::
+
+    logging.disable(logging.CRITICAL)
+
+And then re-enable it by running::
+
+    logging.disable(logging.NOTSET)
+
```

Comment:
I referred [this](https://stackoverflow.com/questions/27647077/fully-disable-python-logging) for documenting the disabling of logging and this was working for me on my local machine. Please LMK if this is the right/standard way.
@rlratzel

Reply:
It might be good to also demonstrate how to disable logging for specific modules, since some applications may want logging elsewhere.  To disable the logging calls in the dispatcher, this can be done:
```python
   logging.getLogger("networkx.utils.backends").setLevel(logging.CRITICAL)
```
Then of course to re-enable debug logging:
```python
   logging.getLogger("networkx.utils.backends").setLevel(logging.DEBUG)
```

For similar reasons, users may want to enable debug logging only for `networkx` and nothing else.  To do that, they can use:
```python
    nxl = logging.getLogger("networkx")
    nxl.addHandler(logging.StreamHandler())
    nxl.setLevel(logging.DEBUG)
```

Reply:
This is starting to sound like we are re-documenting how to do logging. Logging needs to be simple and useful to be included.

Can we point to a good link that explains logging in a clear way? Could that let us remove docs here rather than repeating (and continually updating) what is developed and controlled eslewhere?

Reply:
I agree, that's a good point.  I think backend users and devs might appreciate a short note about how logging can be used to help tune/debug workflows though, so maybe something like this?
```
Add the following code to enable debug log messages:

    import logging
    logging.basicConfig(level=logging.DEBUG)

See https://docs.python.org/3/library/logging.html for detailed usage
and https://docs.python.org/3/howto/logging-cookbook.html for additional examples.
```
If that's too much, maybe just this (or some version of this)?
```
NetworkX has debug logging which can be enabled to assist backend users and developers in
debugging (e.g. what backend is being used and what args are being passed) using the standard
python logging library.  See the Python docs on logging for usage and examples:

https://docs.python.org/3/library/logging.html
https://docs.python.org/3/howto/logging-cookbook.html
```

Reply:
Your first example is certainly not too much: how to turn it on and a pointer to where to learn more. 
And I like the second example's first sentence as an overview.  Is this too much?
```
NetworkX has debug logging which can be enabled to assist backend users and developers in
debugging (e.g. what backend is being used and what args are being passed) using the standard
python logging library. Add the following code to enable debug log messages:

    import logging
    logging.basicConfig(level=logging.DEBUG)

See https://docs.python.org/3/library/logging.html for detailed usage
and https://docs.python.org/3/howto/logging-cookbook.html for additional examples.
```

Reply:
I only added the first link because the second link is very clearly available on the first link's page in a box with the heading "Important". And I also added the code for disabling the logging because that seemed like an important functionality and I thought it would be nice if someone doesn't have to go to an external link to find how it is done.